controllers/nutanixmachine_controller.go

```diff
@@ -187,7 +187,7 @@ func (r *NutanixMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}()
 
 	// Handle deleted machines
-	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+	if !ntxMachine.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
could you please adda. comment in code explaining why do we need to use ntxMachine as compared to machine and impact of that as we were not using it